### PR TITLE
use secrets for container v0

### PR DESCRIPTION
**Pull request by vhvb1989**
This change ensures that connections strings for resource type container.v0 are secured using containerApps secrets and reference from container environment.

fix: https://github.com/Azure/azure-dev/issues/3265

### CODE REVIEW

File: cli/azd/resources/apphost/templates/resources.bicept [hunk 1]
```diff
@@ -162,20 +162,36 @@ resource {{bicepName $name}} 'Microsoft.App/containerApps@2023-05-02-preview' =
...
+      ]
 {{- end}}
     }
     template: {
       containers: [
         {
           image: '{{$value.Image}}'
-          name: '{{$name}}'
-{{- if $value.Env }}
+          name: '{{containerAppName $name}}-${resourceToken}'
```
Comment (ellismg) [inline cli/azd/resources/apphost/templates/resources.bicept:181]: Did we intend to add the ${resourceToken} here?  I'm not sure we need it to uniquify the container name itself inside the template?

**Review by ellismg — COMMENTED** [review]
I think this is correct - but I would love it if we added some coverage to one of the existing tests that generates all the bicep so I could also see how the generated files ended up changing.

File: cli/azd/resources/apphost/templates/resources.bicept (hunk 1 above)
Comment (vhvb1989) [reply cli/azd/resources/apphost/templates/resources.bicept:181]: will remove. This was a mistake

**Comment by vhvb1989** [comment]
> I think this is correct - but I would love it if we added some coverage to one of the existing tests that generates all the bicep so I could also see how the generated files ended up changing.

Tests updated :)
snap tests were actually failing because azd is now adding the secrets section.

File: cli/azd/internal/scaffold/funcs.go [hunk 2]
```diff
@@ -157,6 +157,18 @@ func ContainerAppSecretName(name string) string {
 	return strings.ReplaceAll(strings.ToLower(name), "_", "-")
 }
 
+// ToDotNotation receives a string and if it is on the form of "${inputs['foo']['bar']}" it returns a new string using dot
+// notation, i.e. "${inputs.foo.bar}". Otherwise, the original string is returned.
+func ToDotNotation(s string) string {
```
[the remark this answers is not in the document]
Comment (vhvb1989) [reply cli/azd/internal/scaffold/funcs.go:162]: This is not related to secrets. It is related to a bicep warning:

image

When writing the input path to bicep, we would use dot notation, like:

image

File: cli/azd/pkg/apphost/generate.go [hunk 3]
```diff
@@ -315,7 +315,7 @@ func (b *infraGenerator) LoadManifest(m *Manifest) error {
 		case "project.v0":
```
[the remark this answers is not in the document]
Comment (vhvb1989) [reply cli/azd/pkg/apphost/generate.go:315]: project.v0 `ConnectionStrings__foo` is already handled.

It produces `containerApp.tmpl.yaml` with the mapping to secrets

**Pull request merged**